src/Microsoft.AspNet.Razor/Parser/HtmlMarkupParser.Document.cs

```diff
@@ -107,5 +112,42 @@ private void ScanTagInDocumentContext()
                 tagBlock.Dispose();
             }
         }
+
+        private bool CurrentTagBlockExpectsJavascriptContent()
+        {
+            var blockBuilder = Context.CurrentBlock;
+
+            Debug.Assert(blockBuilder != null);
+
+            var potentialAttributes = blockBuilder.Children
+                .OfType<Block>()
+                .Where(block =>
+                    block.ChunkGenerator is AttributeBlockChunkGenerator &&
+                    block.Children.Count() >= 2);
+
+            if (potentialAttributes.Any())
+            {
+                var typeAttribute = potentialAttributes
+                    .FirstOrDefault(attributeBlock => (attributeBlock.Children.First() as Span)
+                        ?.Content
+                        .TrimStart()
+                        .StartsWith("type", StringComparison.OrdinalIgnoreCase) == true);
+
+                if (typeAttribute != null)
+                {
+                    var contentValues = typeAttribute.Children
+                        .OfType<Span>()
+                        .Where(childSpan => childSpan.ChunkGenerator is LiteralAttributeChunkGenerator)
+                        .Select(childSpan => ((LiteralAttributeChunkGenerator)childSpan.ChunkGenerator))
+                        .Select(valueGenerator => valueGenerator.Prefix.Value + valueGenerator.Value.Value);
```

Comment:
To protect against cases such as: `<script type="text /html">` Or enable cases such as: `<script type="  text/html">`